test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java

```diff
@@ -2612,4 +2622,59 @@ public void testProfilesIncludesDefault() {
         assertEquals(new HashSet<>(Arrays.asList("default", "test")), profileSettings.stream().map(s -> s.profileName).collect(Collectors
             .toSet()));
     }
+
+    public void testChannelCloseWhileConnecting() throws IOException {
+        /*
+         * We have to keep the first connection open so the handshake succeeds and then we later fail when checking if all the connections
+         * are open. For mock TCP transport there is only a single connection that is open so we have to defer closing it until immediately
+         * after the handshake is complete to simulate a channel closing while we are connecting.
+         */
+        final MockTransportService service;
+        if (opensMultipleChannels()) {
+            final AtomicBoolean first = new AtomicBoolean();
+            service =
+                    buildService("service", version0, clusterSettings, Settings.EMPTY, true, true, channel -> {
+                        if (!first.compareAndSet(false, true)) {
```

Comment:
I understand that this works. But the booleans (as labelled) seem backwards to me.

Isn't it:

```
final AtomicBoolean first = new AtomicBoolean(true)
...
if (!first.compareAndSet(true, false)) {
}
```

?

Reply:
I pushed 53a9b2c0e666c4d9c6962af74009b0562f11a776.